frontend/src/components/Dish/DishIndex.jsx

```diff
@@ -1,49 +1,41 @@
-import React, { useEffect, useState, useContext } from "react";
+import React, { useContext } from "react";
 import DishItem from "./DishItem";
 import userContext from "../../contexts/context";
 
 const _ = require("lodash");
 
-const apiUtil = require("../../util/apiUtil");
+function DishIndex({ setFormInfo, dishesState }) {
+  const { toggleModal, language, isAdmin } = useContext(userContext);
 
-function DishIndex(props) {
-  const [dishesState, setDishes] = useState(() => []);
-  const isAdmin = useContext(userContext).admin;
+  function preperForm(formType, dish) {
+    setFormInfo({
+      entityName: "dish",
+      entity: dish,
+      postOrPut: formType,
+    });
+    toggleModal();
+  }
 
-  function getDishesSucces(dishes) {
-    let array = [];
-    array.push(
-      <DishItem
-        isAdmin={isAdmin}
-        formType={"new"}
-        setFormInfo={props.setFormInfo}
-        setFormCallback={props.setFormCallback}
-      />
-    );
-    _.forEach(dishes, (dish) => {
-      array.push(
+  return (
+    <div>
+      {isAdmin ? (
```

Comment:
I think any logic that relates to `isAdmin` and create/edit of entities should be generic and in `Page` (or `Navbar`)